content/blog/non-decimal-numbers-in-tech/index.md

```diff
@@ -0,0 +1,225 @@
+
+Given this information, how would we represent the number **`50`**?
+
+![The hexadecimal number "032" which shows 0 "two hundred sixes", 3 "sixteens", and 2 "ones" to combine together to make 50. See the below explaination for more](./base_16_50.svg)
+
+Assuming we have a _ones_ column, a _sixteens_ column, and a _two-hundred fifty sixths_, let's do a similar method of calulating the number as we did with binary.
+
+Is `256` less than or equal to `50`? No. That's a **`0`**
+Is `16 <= 50`? Yes. So we know it's _at least_ _`1`_.
+
+Now, how many times can you put `16` in `50`?
+
+`16 * 2 = 32` and `32 <= 50`, so it's _at least_ _`2`_
+
+`16 * 3 = 48` and `48 <= 50` so it's _at least_ _`3`_
+
+`16 * 4 = 64`. However, `64 > 50`, therefor the _sixteenth_ place cannot be _`4`_, therefore it must be **`3`**
+
+So now that we know the most we can have in the _sixteenth_ place, we can subtract the sum (`48`) from our result (`50`)
+
+`50 - 48 = 2`
+
+Now onto the _ones_ place:
+
+How many _ones_ can fit into _`2`_?
+
+`1 * 1 = 1` and `1 <= 2`, so it's _at least_ _`1`_
+
+`1 * 2 = 2` and `2 <= 2` and because these number equal, we know that there must be **`2`** _twos_.
+
+Now if we add up these numbers:
+
+| Column | Value   |
+| ------ | ------- |
+| `256`  | **`0`** |
+| `16`   | **`3`** |
+| `1`    | **`2`** |
+
+## Why `256`?
+
+While reading through this, you may wonder "Where did the `256` come from?". Let's take a step back to anaylize this question.
+
+If you recall, we use the 15 symbols in hexadecimal:
+
+`0`, `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `A`, `B`, `C`, `D`, `E`, `F`
+
+If we reflect these numbers in a single digit (1 number column), the biggest number we can reflect is `F`: `15`.
+
+This is similar to how the biggest number we can reflect with a single digit in the decimal system is `9`.
+
+In order to add a number larger than `15` in the hexadecimal system, we need to add another decimal/column. This column would represent the _sixteens_ place. Having `F` in this column, the highest we can represent, and an `F` in the _ones_ would allow us to have `FF`: `255`. As a result, we need to add a _two-hundred-fifty-six_ column to represent any numbers higher.
+
+> For those that have experience in algebra, you'll notice that these are all exponents of each other.
+>
+> Just as _`100`_ is `10^2` for the decimal system, `256` is `16^2`. We can follow this pattern to the next number in the hexadecimal column: `4096`, which is `16^3`. You can even apply it to `1` which is `16^0`
+>
+> Binary works in the same manor. The first 5 columns/digits of binary are: `1`, `2`, `4`, `8`, `16` . These numbers align respectively to their binary exponents: `2^0`, `2^1`, `2^2`, `2^3`, `2^4`
+
```

Comment:
It might be interesting to show how a number like '732' for example in base 10 can be written as 7*10^2 + 3*10^1 + 2*10^0. And just like in something like octal the number '732' would be represented as 7*8^2 + 3*8^1 + 3*8^0. This is just being annoying but I always found this interesting.